### PR TITLE
Fix Ampcontext variables in singlepass

### LINKED ISSUE #20002: Fix amp-context API for single pass.

### PR DESCRIPTION
**Pull request by zhouyx**
Since ad script within iframe can access the variables, I believe we should add the vars to externs. 
Let me know if this is the correct fix. 

One thing that confuses me is that does it mean we should add every var and method from ampcontext.js to the extern list. 

to @erwinmombay @lannka 

Closes #20002

### CODE REVIEW

**Comment by lannka** [comment]
pls hold this PR for the test unmute: #16828
so that we can catch all of them in one shot.

**Comment by zhouyx** [comment]
Passing the JsonObject fix the issue.  PTAL. 

@erwinmombay please let me know if it is safe to remove the externs we have. Thanks.

File: build-system/amp.extern.js [hunk 1]
```diff
@@ -144,6 +144,9 @@ process.env.SERVE_MODE;
 window.IS_AMP_ALT;
 
 // Exposed to ads.
+// TODO: @erwinmombay I do believe this is safe to remove because
+// All are passed to ampcontext in JsonObject. While isMaster and master is
+// calculated in ampcontext-integration within the iframe.
```
Comment (erwinmombay) [inline build-system/amp.extern.js]: you'd have to change all the reads to use bracket notation if you want to remove this

Comment (zhouyx) [reply build-system/amp.extern.js]: Here https://github.com/ampproject/amphtml/blob/master/3p/ampcontext.js#L294?

Comment (zhouyx) [reply build-system/amp.extern.js]: Asked @erwinmombay offline. We want to leave the extern for future optimization that is not related to singlepass. I removed my comment. We will need to add missing variables to the externs, or use bracket notation.

**Pull request merged**